Order view - Refresh shipping tab when product is added, removed or updated

sowbiba opened the pull request:
<!-----------------------------------------------------------------------------
Thank you for contributing to the PrestaShop project! 

Please take the time to edit the "Answers" rows below with the necessary information.

Check out our contribution guidelines to find out how to complete it:
https://devdocs.prestashop.com/1.7/contribute/contribution-guidelines/#pull-requests
------------------------------------------------------------------------------>

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.7.x
| Description?  | Refresh shipping tab when product list changed
| Type?         | bug fix
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #21267 
| How to test?  | See description of #21267. Test cases when a product is added, removed or has quantity, price updated. Test cases with cart rules

<!-- Click the form's "Preview" button to make sure the table is functional in GitHub. Thank you! -->

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/prestashop/prestashop/22367)
<!-- Reviewable:end -->

sowbiba committed:
Order view - Refresh shipping tab when product is added or removed

PierreRambaud commented on admin-dev/themes/new-theme/js/pages/order/view/order-shipping-refresher.js:
```diff
@@ -0,0 +1,44 @@
+    }
+
+    refresh(orderId) {
+        $.ajax(this.router.generate('admin_orders_get_shipping', {orderId}))
```
```suggestion
        $.getJSON(this.router.generate('admin_orders_get_shipping', {orderId}))
```

sowbiba committed:
Replace $.ajax by $.getJSON

SD1982 commented:
Hi @sowbiba 
Everything looks good to me except for this little point:
With you PR when we edit an order the "payment block" is not correctly updated anymore without a page refresh (it was before the PR)

**screenrecords**
Without PR:
https://drive.google.com/file/d/1RGEqtwpdK9MO-Mz5dxaUr9PGAE5g8XzT/view?usp=sharing
With PR:
https://drive.google.com/file/d/1xza1Zq6EiJm_dK4reWkOfxf7SVDSE7fn/view?usp=sharing

PierreRambaud commented on admin-dev/themes/new-theme/js/pages/order/view/order-shipping-refresher.js:
```diff
@@ -0,0 +1,44 @@
+
+import Router from '@components/router';
+import OrderViewPageMap from '@pages/order/OrderViewPageMap';
+import InvoiceNoteManager from '../invoice-note-manager';
```
```suggestion
import InvoiceNoteManager from '@pages/order/invoice-note-manager';
```
Avoid relatives paths

sowbiba replied:
Not used anyway. Bad copy/paste. I removed the import ;)

PierreRambaud commented on src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php:
```diff
@@ -809,6 +809,24 @@ public function getDocumentsAction(int $orderId)
     }
 
+    /**
+     * @AdminSecurity("is_granted(['create', 'update'], request.get('_legacy_controller'))", redirectRoute="admin_orders_index")
```
Should be `view` because of `GetOrderForViewing` otherwise, something is confusing

sowbiba replied:
I fix the other 2 actions to use `read`
Nice catch

sowbiba committed:
Fix security rule for Controller action

sowbiba committed:
Use $.ajax instead of $.getJSON in order-payments-refresher

sowbiba commented:
> Hi @sowbiba
> Everything looks good to me except for this little point:
> With you PR when we edit an order the "payment block" is not correctly updated anymore without a page refresh (it was before the PR)
> 
> **screenrecords**
> Without PR:
> https://drive.google.com/file/d/1RGEqtwpdK9MO-Mz5dxaUr9PGAE5g8XzT/view?usp=sharing
> With PR:
> https://drive.google.com/file/d/1xza1Zq6EiJm_dK4reWkOfxf7SVDSE7fn/view?usp=sharing

Fixed

It's because I changed $.ajax by $.getJSON

```
$.ajax(
  .then(
     response => {}, //case success
     response => {}, //case error
  )
)
```
$.getJSON doesn't work this way
You have to specify

```
$.getJSON()
.done()
.fail()
```

SD1982 commented:
LGTM now !!
Tested :

- initial issue
- add product
- delete product
- edit product quantity
- edit product price
- add discount ( % and amount)
- add free shipping

Thanks @sowbiba !!

matks commented:
Merge can happen after conflicts are solved 😉  ping @sowbiba

sowbiba committed:
Rebase 1.7.7.x

sowbiba commented:
Rebase done

Progi1984 reviewed (changes requested):
## DONT MERGE BEFORE RELEASE 1.7.7.1

Progi1984 commented:
Thanks @sowbiba & @SD1982
